src/Tools/Source/RunTests/Options.cs

```diff
@@ -167,6 +167,24 @@ internal class Options
                 return null;
             }
 
+            artifactsPath ??= TryGetArtifactsPath();
+            dotnetFilePath ??= Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), @"dotnet\dotnet.exe");
```

Comment:
If RunTests is started via dotnet CLI do we want to try and figure out where the dotnet that started RunTests is located instead of just looking in Program Files?

Reply:
Yes absotulely. That was on my "todo" list for this change. Given that we will always be using `dotnet` to invoke this tool (it won't have an app host) we can use this logic 

https://github.com/dotnet/runtime/issues/38462#issuecomment-653620531